patch/kconfig-inclusions

```diff
@@ -83,7 +81,6 @@ CONFIG_MLXREG_IO=m
 CONFIG_MLX_WDT=m
 CONFIG_MLXREG_LC=m
 CONFIG_SENSORS_MLXREG_FAN=m
-CONFIG_GPIO_GENERIC=m
```

Comment:
this seems generic module, why are we remove GPIO_GENERIC AND REGMAP_I2C?

Reply:
will these removal affect other platforms? @saiarcot895 ?

Reply:
In the CI builds, both `CONFIG_GPIO_GENERIC` and `CONFIG_REGMAP_I2C` are still enabled, and set to `m`, likely due to other dependencies, so removing those two configs in this PR has no net impact.

Reply:
Both the Kconfig's are removed since they are redundant as saikrishna already pointed out. No harm in removing them from here.

Reply:
thanks for confirmation.